Automattic/wp-calypso
# Directly: Add state to Redux

This PR introduces the Directly RTM widget's `status` into the Redux state tree. The purpose is to allow components to respond to changes in Directly's status. For example in this PR, on the `/help/contact` page we're now falling back to forum support if Directly fails to load.

### To test

Sign in as a user with no paid upgrades and head to http://calypso.localhost:3000/help/contact. You should see the form with a submit button saying "Ask an Expert" (this is the Directly integration form).

Open Redux Dev Tools. You should be able to see both a `DIRECTLY_INITIALIZATION_START` and a `DIRECTLY_INITIALIZATION_SUCCESS` action dispatched. You should also see the current status under `state.help.directly.status`.

**To test a failed lib load** change [the library URL in `lib/directly`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/directly/index.js#L18) to a bogus address. Refresh the page and you should see the forums form appear instead. Redux Dev Tools should show a `DIRECTLY_INITIALIZATION_START` and a `DIRECTLY_INITIALIZATION_ERROR` action, and the state tree should reflect the error.
